File: Java/SivitasAkademik.java
/*Saya Muhammad Daffa Yusuf Fadhilah dengan NIM 2100543mengerjakan evaluasi Latihan 
dalam mata kuliah Design Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti
yang telah dispesifikasikan. Aamiin. */

//Berikut class untuk 'SivitasAkademik' yang menjadi turunan dari class 'Mahasiswa'
public class SivitasAkademik extends Mahasiswa{
    private String asal_universitas;
    private String email_edu;
    
    //Empty Constructor untuk 'Mahasiswa'
    public SivitasAkademik() {
        super("", "", '-', "", "", "");
        asal_universitas = "";
        email_edu = "";
    }
    
    //Parameter Constructor untuk 'Mahasiswa'
    public SivitasAkademik(String NIK, String nama, char jenis_kelamin, String NIM, String prodi, String fakultas,
            String asal_universitas, String email_edu) {
        super(NIK, nama, jenis_kelamin, NIM, prodi, fakultas);
        this.asal_universitas = asal_universitas;
        this.email_edu = email_edu;
    }

    //Setter dan Getter dari SivitasAkademik
    public void set_asal_universitas(String asal_universitas) {
        this.asal_universitas = asal_universitas;
    }

    public void set_email_edu(String email_edu) {
        this.email_edu = email_edu;
    }

    public String get_asal_universitas() {
        return asal_universitas;
    }

    public String get_email_edu() {
        return email_edu;
    }
}

File: C++/SivitasAkademik.cpp
/*Saya Muhammad Daffa Yusuf Fadhilah dengan NIM 2100543 mengerjakan evaluasi Latihan 2
dalam mata kuliah Design Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti
yang telah dispesifikasikan. Aamiin. */

//Include untuk penggunaan string dan class Mahasiswa

#include <iostream>
#include <string>
#include "Mahasiswa.cpp"

//Deklarasi std agar tidak perlu menulisnya kembali pada code
using namespace std;



class SivitasAkademik : public Mahasiswa
{
protected:

    ///Membuat atribut yang dibutuhkan/ditambah dari parent
    string asal_universitas;
    string email_edu;

public:

    //Fungsi construct
    SivitasAkademik(){
        this->NIK = "";
        this->nama = "";
        this->jenis_kelmain = '-';
        this->NIM = "";
        this->prodi = "";
        this->fakultas = "";
        this->asal_universitas = "";
        this->email_edu = "";
    }

    //Fungsi construct dengan parameter
    SivitasAkademik(string NIK, string nama, char jenis_kelamin, string NIM, string prodi, string fakultas, string asal_universitas, string email_edu){
        this->NIK = NIK;
        this->nama = nama;
        this->jenis_kelmain =  jenis_kelamin;
        this->NIM = NIM;
        this->prodi = prodi;
        this->fakultas = fakultas;
        this->asal_universitas = asal_universitas;
        this->email_edu = email_edu;
    }

    //Setter dan getter tiap attribut yang ada pada class ini
    void set_asal_universitas(string asal_universitas){
        this->asal_universitas = asal_universitas;
    }

    void set_email_edu(string email_edu){
        this->email_edu = email_edu;
    }

    string get_asal_universitas(){
        return this->asal_universitas;
    }
    
    string get_email_edu(){
        return this->email_edu;
    }

    ~SivitasAkademik(){

    }
};
